crates/price-feed/src/price_provider/dia.rs

```diff
@@ -0,0 +1,27 @@
+use super::{PriceFetchError, PriceInfoFromProvider};
+
+pub(crate) async fn fetch_dia(url: &str) -> Result<PriceInfoFromProvider, PriceFetchError> {
+    Ok(reqwest::get(url)
+        .await?
+        .json::<PriceInfoFromProvider>()
+        .await?)
+}
+
+#[cfg(test)]
```

Comment:
Nit: this test is really an integration test, since it requires some external resource to be available. I'd put it in `crates/price-feed/tests/` directory